nanachan/extensions/profiles.py

```diff
@@ -119,229 +104,328 @@ async def promo(self, ctx: Context):
-
-        message = False
-        if unknown_members:
-            message = True
-            for page in list_display('Unknown members', unknown_members):
-                await ctx.send(page)
-        if members_without_pp:
-            message = True
-            for page in list_display('Members without profile picture', members_without_pp):
-                await ctx.send(page)
-        if not message:
-            await ctx.send('```All this server\'s members are known and have a profile picture```')
-
-    @staticmethod
-    async def _fetch_vcard(message):
-        for attachment in message.attachments:
-            if attachment.filename.endswith('.vcf'):
-                break
-        else:
-            raise BadArgument('`.vcf` file not found')
+        await AutoNavigatorView.create(
+            self.bot,
+            ctx.reply,
+            title='ENSEEIHT members',
+            description='\n'.join(text),
+            author_name=str(guild),
+            author_icon_url=icon_url,
+            footer_text=f'{len(text)} members',
+        )
 
-        return vobject.readOne((await attachment.read()).decode())  # type: ignore |not exported ig
+    async def _get_profile(self, discord_id: int):
+        profile_resp = await get_nanapi().user.user_get_profile(discord_id)
+        if not success(profile_resp):
+            raise RuntimeError(profile_resp.result)
+        return profile_resp.result
 
     @staticmethod
-    async def _create_or_update_profile(member: Member, vcard):
-        full_name = vcard.fn.value
-        promotion = re.sub('&amp,', '&', Profiles._flatten(vcard.org.value))
-        telephone = vcard.tel.value if 'tel' in vcard.contents else ''
-        photo = base64.b64encode(vcard.photo.value).decode('ascii') if len(
-            vcard.photo.value) > 5 else ''
-
-        resp = await get_nanapi().user.user_upsert_profile(
-            member.id,
-            UpsertProfileBody(discord_username=str(member),
-                              full_name=full_name,
-                              promotion=promotion,
-                              telephone=telephone,
-                              photo=photo))
+    async def _create_or_update_profile(member: Member | discord.User, payload: UpsertProfileBody):
+        resp = await get_nanapi().user.user_upsert_profile(member.id, payload)
         if not success(resp):
             raise RuntimeError(resp.result)
         profile = resp.result
         return profile
 
     @staticmethod
-    def _flatten(l):
-        if isinstance(l, list):
-            result = Profiles._flatten(l[0])
-            if len(l) > 1:
-                for element in l[1:]:
-                    result += ',' + Profiles._flatten(element)
-            return result
-        else:
-            return l
-
-    @staticmethod
-    async def _send_vcard(send_func, member: Optional[Member], profile):
+    def create_vcard(member: Optional[Member], profile: ProfileSearchResult):
```

Comment:
`Optional[x]` → `x | None`
Optional is deprecated and we only target the latest version of Python so we don’t have to care about compatibility with Python <3.10